sys/include/string_utils.h

```diff
@@ -70,6 +72,23 @@ static inline void explicit_bzero(void *dest, size_t n_bytes)
 }
 #endif
 
+/**
+ * @brief Copy the string, or as much of it as fits, into the dest buffer.
+ *
+ * Preferred to `strncpy` since it always returns a valid string, and doesn't
+ * unnecessarily force the tail of the destination buffer to be zeroed.
+ * If the zeroing is desired, it's likely cleaner to use `strscpy` with an
+ * overflow test, then just memset the tail of the dest buffer.
+ *
+ * @param[out] dest     Where to copy the string to
+ * @param[in]   src     Where to copy the string from
+ * @param[in] count     Size of destination buffer
```

Comment:
```suggestion
 * @param[in] count     Size of destination buffer
 *
 * @pre       The destination buffer is at least one byte large, as
 *            otherwise the terminating zero byte won't fit
```